hudi-common/src/main/java/org/apache/hudi/common/table/timeline/LSMTimeline.java

```diff
@@ -158,13 +159,18 @@ public static int latestSnapshotVersion(HoodieTableMetaClient metaClient) throws
         LOG.warn("Error reading version file {}", versionFilePath, e);
       }
     }
+
     return allSnapshotVersions(metaClient).stream().max(Integer::compareTo).orElse(-1);
   }
 
   /**
    * Returns all the valid snapshot versions.
    */
   public static List<Integer> allSnapshotVersions(HoodieTableMetaClient metaClient) throws IOException {
+    StoragePath archivedFolderPath = new StoragePath(metaClient.getArchivePath());
```

Comment:
I search around the invokers, one is archive `LSMTimelineWriter#clean` and another is `LSMTimeline#`latestSnapshotVersion, both would ensure the archving already happens, is this a case specific for backward compatibility?

Reply:
Yes, found during testing the 0.14.x writer with 1.x reader testing for a newly created 0.14.x table. LSMTimeline#latestSnapshotVersion was calling LSMTimeline#allSnapshotVersions due to the absence of latest snapshot version file.

Reply:
Got it, thanks for thee explaination.